play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -738,14 +767,19 @@ export class ChatTextEntry extends preact.Component<{
 			return this.toggleFormatChar('*');
 		} else if (ev.keyCode === 192 && cmdKey) { // Ctrl + ` key
 			return this.toggleFormatChar('`');
-		// } else if (e.keyCode === 9 && !e.ctrlKey) { // Tab key
-		// 	const reverse = !!e.shiftKey; // Shift+Tab reverses direction
-		// 	return this.handleTabComplete(this.$chatbox, reverse);
+		} else if (ev.keyCode === 9 && !ev.ctrlKey) { // Tab key
+			const reverse = !!ev.shiftKey; // Shift+Tab reverses direction
+			return this.handleTabComplete(reverse);
 		} else if (ev.keyCode === 38 && !ev.shiftKey && !ev.altKey) { // Up key
 			return this.historyUp(true);
 		} else if (ev.keyCode === 40 && !ev.shiftKey && !ev.altKey) { // Down key
 			return this.historyDown(true);
 		} else if (ev.keyCode === 27) { // esc
+			if (this.undoTabComplete()) {
+				ev.preventDefault();
+				ev.stopPropagation();
```

Comment:
```suggestion
```

These are unnecessary, `return true` does the same thing.

Reply:
Also, for future reference, always use `stopImmediatePropagagion` instead of `stopPropagation`, unless for some reason you actually want to stop some but not all propagation.